test/datadog/fips_mode_test.go

```diff
@@ -0,0 +1,91 @@
+package datadog
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/helm-charts/test/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestFIPSModeConditions(t *testing.T) {
+	tests := []struct {
+		name                   string
+		setFipsEnabledSetting  bool
+		setUseFipsImageSetting bool
+		expectFipsProxy        bool
+		expectFipsImageSuffix  bool
+	}{
+		{
+			name:                   "fips.useFipsImages true should not use fips-proxy and use fips image",
+			setFipsEnabledSetting:  true,
+			setUseFipsImageSetting: true,
+			expectFipsProxy:        false,
+			expectFipsImageSuffix:  true,
+		},
+		{
+			name:                   "fips.useFipsImages false and fips.enabled true should use fips-proxy and not use fips image",
+			setFipsEnabledSetting:  true,
+			setUseFipsImageSetting: false,
+			expectFipsProxy:        true,
+			expectFipsImageSuffix:  false,
+		},
+		{
+			name:                   "fips.useFipsImages false and fips.enabled false should not use fips-proxy or fips image",
+			setFipsEnabledSetting:  false,
+			setUseFipsImageSetting: true,
+			expectFipsProxy:        false,
+			expectFipsImageSuffix:  false,
+		},
+		{
+			name:                   "fips.useFipsImages false and fips.enabled false should not use fips-proxy or fips image",
+			setFipsEnabledSetting:  false,
+			setUseFipsImageSetting: true,
+			expectFipsProxy:        false,
+			expectFipsImageSuffix:  false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			values := map[string]string{
+				// "fips.useFipsImages": strconv.FormatBool(tt.setUseFipsImageSetting),
+				// "fips.enabled":       strconv.FormatBool(tt.setFipsEnabledSetting),
+			}
+
+			manifest, err := common.RenderChart(t, common.HelmCommand{
+				ReleaseName: "datadog",
+				ChartPath:   "../../charts/datadog",
```

Comment:
```suggestion
				ChartPath:   "../../charts/datadog",
				ShowOnly:    []string{"templates/daemonset.yaml"},
```
This will render only Daemonset which can be correctly unmarshalled. Without this setting, whole chart gets rendered which is couple dozen resources so they can't me unmarshalled in structs.